File: class_primer/class_primer__static_member.cs
using System.Linq;
using static System.Console;

internal class Program
{
    class Cus
    {
        internal uint Amo { get; set; } = 0;

        internal virtual void Order( string s, uint i )
        {
            if( StrComp( s, "food" ) || StrComp( s, "softdrink" ) ) {
                Amo += i;
            }
            return;
        }
    }

    class Adu : Cus
    {
        bool Dis { get; set; } = false;

        internal override void Order( string s, uint i )
        {
            if( StrComp( s, "food" ) == true ) {
                Amo += Dis ? i - 200 : i;
            } else if( StrComp( s, "softdrink" ) == true ) {
                Amo += i;
            } else if( StrComp( s, "alcohol" ) == true ) {
                Amo += i;
                Dis = true;
            }
            return;
        }
    }

    static void Main()
    {
        int c = 0;
        int[] nk = ReadLine().Split().Select( int.Parse ).ToArray();
        Cus[] a = new Cus[nk[0]];

        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        foreach( int i in Enumerable.Range( 0, nk[0] ) ) {
            a[i] = int.Parse( ReadLine() ) < 20 ? new Cus() : new Adu();
        }
        foreach( int i in Enumerable.Range( 0, nk[1] ) ) {
            string[] s = ReadLine().Split();
            int n = int.Parse( s[0] ) - 1;

            if( s.Length == 3 ) {
                a[n].Order( s[1], uint.Parse( s[2] ) );
            } else if( StrComp( s[1], "0" ) == true ) {
                a[n].Order( "alcohol", 500 );
            } else if( StrComp( s[1], "A" ) == true ) {
                c++;
                WriteLine( a[n].Amo );
            }
        }
        WriteLine( c );
        Out.Flush();
        return;
    }

    static bool StrComp( string a, string b )
    {
        return string.Compare( a, b, System.StringComparison.Ordinal ) == 0;
    }
}

File: class_primer/class_primer__static_member.py
class cus:
    def __init__(self):
        self.amo = 0

    def order(self, s, i):
        if s == "food" or s == "softdrink":
            self.amo += i

    def getAmount(self):
        return self.amo

class adu(cus):
    def __init__(self):
        super().__init__()
        self.dis = False

    def order(self, s, i):
        if s == "food":
            self.amo += i - 200 if self.dis else i
        elif s == "softdrink":
            self.amo += i
        elif s == "alcohol":
            self.amo += i
            self.dis = True

n, k = list(map(int, input().split()))
o = 0
l = []

for _ in range(n):
    a = int(input())

    l.append(cus() if a < 20 else adu())
for _ in range(k):
    s = input().split()

    if len(s) == 3:
        l[int(s[0]) - 1].order(s[1], int(s[2]))
    elif s[1] == "0":
        l[int(s[0]) - 1].order("alcohol", 500)
    elif s[1] == "A":
        o += 1
        print(cus.getAmount(l[int(s[0]) - 1]))
print(o)
